torch/_dynamo/variables/user_defined.py

```diff
@@ -142,14 +143,16 @@ def var_getattr(self, tx, name: str) -> "VariableTracker":
         if self.value is collections.OrderedDict and name == "fromkeys":
             return super().var_getattr(tx, name)
 
-        if name in getattr(self.value, "__dict__", {}) or (
+        if ConstantVariable.is_literal(obj):
+            return ConstantVariable.create(obj)
+        elif isinstance(obj, enum.Enum):
+            return EnumVariable(obj)
+        elif name in getattr(self.value, "__dict__", {}) or (
```

Comment:
The constant/enum handling should take precedence over the other cases, since they would return a Dynamo variable no matter if there is ```source```.